### DISCUSSION TITLE
Questions around `loading.tsx`

### DISCUSSION THREAD

**Discussion by fnb-software** [vercel/next.js#55436] [App Router]
1. Why is the `loading` inherited for nested routes? A loading screen can be really specific to the current page, not the nested ones.

2. Why is the `loading` displayed when navigating to a route that is not `async`? Surely `next` could also optimize that to have the html ready to be displayed immediately?

3. What is the trade-off? When is the `loading` loaded? Otherwise I could simply use `loading` and not `page` to have a faster website.

Thanks!

**Comment by tdonhauser** [comment]
I have another question added to the ones you mentioned:
How can I use loading.tsx if I have dynamic routes. Like with i18n when the path is dependend on the language like this:

./app/[lang]/loading.tsx

I would like to retrieve the lang, but params are not delivered to layout.tsx
If I try to use useRouter I need to convert the hole component to a client component. Is there another way?

**Reply by tkrotoff** [reply]
Same here, I expected loading.tsx and error.tsx to take [`params` like layout.tsx and page.tsx](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#convention)
